app/app.go

```diff
@@ -526,14 +528,20 @@ func wireCoreWorkflow(ctx context.Context, life *lifecycle.Manager, conf Config,
 
 	retryer := retry.New[core.Duty](deadlineFunc)
 
-	cons, startCons, err := newConsensus(cluster, tcpNode, p2pKey, sender,
-		deadlinerFunc("consensus"), gaterFunc, consensusDebugger.AddInstance)
+	consensusFactory, err := consensus.NewConsensusFactory(tcpNode, sender, peers, p2pKey, deadlinerFunc, gaterFunc, consensusDebugger)
 	if err != nil {
 		return err
 	}
 
+	defaultConsensus := consensusFactory.DefaultConsensus()
+	startDefaultConsensus := lifecycle.HookFuncCtx(defaultConsensus.Start)
+
+	coreConsensus := consensusFactory.CurrentConsensus() // points to DefaultConsensus() initially
```

Comment:
`defaultConsensus` is always used by Priority/InfoSync and cannot be changed for this purpose. This is because Priority is reaching consensus on what "core" consensus protocol must be running.
`coreConsensus` is dynamic and can be switched at runtime; the exact protocol is selected by Priority. Before that selection happens (this takes time), "core" will continue using the default consensus.

Reply:
Got it, thank you!